cli/lock.ml

```diff
@@ -109,6 +109,43 @@ let root_depexts local_opam_files =
     (fun _pkg (_version, opam_file) acc -> OpamFile.OPAM.depexts opam_file :: acc)
     local_opam_files []
 
+let root_pin_depends local_opam_files =
+  OpamPackage.Name.Map.fold
+    (fun _pkg (_version, opam_file) acc -> OpamFile.OPAM.pin_depends opam_file @ acc)
+    local_opam_files []
+
+let pull_pin_depends (pin_depends : (OpamPackage.t * OpamUrl.t) list) =
+  (* TODO: Group pin-depends with the same url. *)
```

Comment:
Can this be taken care of or a ticket created for this?

Reply:
I implemented it and removed the comment!